test/_init_tests.js

```diff
@@ -214,14 +214,35 @@ it = function(message, runnable) { // eslint-disable-line no-native-reassign, no
     return realIt(message, runnable);
   }
   return realIt(message, done => {
```

Comment:
We can refactor this so Promises take care of everything:

```js
// Adapted from Babel's asyncToGenerator
return realIt(message, () => {
  return new Promise((resolve, reject) => {
    const gen = runnable();
    function step(key, arg) {
      let info;
      try {
        info = gen[key](arg);
      } catch (error) {
        reject(error);
        return;
      }

      const { value, done } = info;
      if (done) {
        resolve(value);
      } else {
        Promise.resolve(value).then(_next, _throw);
      }
    }
    function _next(value) {
      step("next", value);
    }
    function _throw(err) {
      step("throw", err);
    }
    _next();
  });
});
```

Differences include:
- Promises handle async (through assimilation)
- Yielded promises can reject, going into the generators throw path
- We avoid using `done` https://github.com/ampproject/amphtml/issues/8966
- Less closures